buildSrc/private/src/main/kotlin/androidx/build/PrepareSqliteSourcesTask.kt

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.build
+
+import com.android.utils.appendCapitalized
+import java.io.File
+import java.net.URL
+import javax.inject.Inject
+import org.gradle.api.Action
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.work.DisableCachingByDefault
+
+/**
+ * Finds the sqlite sources and puts them into [destinationDirectory].
+ *
+ * This task runs differently between AOSP and Github (Playground).
+ *
+ * On AOSP, the sources are in an external prebuilts repository and simply get copied into the given
+ * [destinationDirectory].
+ * On Github, they are downloaded from SQLite servers.
+ *
+ * To ensure each version is consistent, we use the [sqliteVersion] parameter and check the SQLite
+ * source code for them.
+ */
+@DisableCachingByDefault(
+    because = "This is a simple copy/validation task that is not worth caching"
+)
+abstract class PrepareSqliteSourcesTask @Inject constructor(
```

Comment:
i'm not sure if this should live in public or private buildSrc.
it is only used by the sqlite-bundled task so we could consider moving this into that build.gradle file but the logic is not straightforward, hence moved into buildSrc.

Reply:
this is super sqlite specific and not even used in aosp, i'd say we move this to sqlite build.gradle

Reply:
done, moved to a groovy file.